### PR TITLE
MINOR: Update build.gradle and release.py to upload streams-scala_2.12

### PR DESCRIPTION
**Pull request by rajinisivaram**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by ijuma — COMMENTED** [review]
Hmm, why isn't uploadArchivesAll working? We should not need the 2_12 version anymore.

**Comment by ijuma** [comment]
For reference, Scala 2.12 was added to the *All commands when we dropped Java 7 support: https://github.com/apache/kafka/commit/e70a191d3038e00790aa95fbd1e16e78c32b79a4#diff-6512f838e273b79676cac5f72456127f

Looks like the update to release.py was incomplete and the supposedly redundant upload archive line was not removed.

**Comment by rajinisivaram** [comment]
@ijuma Thanks for the review. The various `xxxAll` tasks in `build.gradle` seem to explicitly run tasks for `core`, which is possibly why `streams-scala` didn't get uploaded for `scala 2.12`. I have removed the explicit `2.12` upload in `release.py` and updated `build.gradle` to run tasks for `streams-scala` as well wherever it was running tasks for `core` with different scala versions. Can you take another look? Thanks.

File: build.gradle [hunk 1]
```diff
@@ -473,28 +473,30 @@ task reportCoverage(dependsOn: ['jacocoRootReport', 'core:reportCoverage'])
...
     startParameter.projectProperties += [scalaVersion: "${sv}"]
-    tasks = ['core:test']
+    tasks = ['core:test', 'streams:streams-scala:test']
   }
 
   tasks.create(name: "srcJar_${taskSuffix}", type: GradleBuild) {
     startParameter = project.getGradle().getStartParameter().newInstance()
     startParameter.projectProperties += [scalaVersion: "${sv}"]
+    // core has a separate source jar for each scala version, streams-scala source is included
+    // in the single streams sources jar
```
Comment (ijuma) [inline build.gradle]: @guozhangwang @mjsax The behaviour with regards to the Scala source mentioned in the comment above is a bit weird.

File: build.gradle [hunk 2]
```diff
@@ -525,20 +527,20 @@ def withDefScalaVersions(taskName) {
...
-tasks.create(name: "jarAll", dependsOn: withDefScalaVersions('jar_core') + pkgs.collect { it + ":jar" }) { }
+tasks.create(name: "jarAll", dependsOn: withDefScalaVersions('jar_scala') + pkgs.collect { it + ":jar" }) { }
 
 tasks.create(name: "srcJarAll", dependsOn: withDefScalaVersions('srcJar') + pkgs.collect { it + ":srcJar" }) { }
 
 tasks.create(name: "docsJarAll", dependsOn: withDefScalaVersions('docsJar') + pkgs.collect { it + ":docsJar" }) { }
 
 tasks.create(name: "testConnect", dependsOn: connectPkgs.collect { it + ":test" }) {}
-tasks.create(name: "testAll", dependsOn: withDefScalaVersions('test_core') + pkgs.collect { it + ":test" }) { }
+tasks.create(name: "testAll", dependsOn: withDefScalaVersions('test_scala') + pkgs.collect { it + ":test" }) { }
```
Comment (ijuma) [inline build.gradle]: Maybe we should use camel case here and a few lines above for consistency with the other task names.

**Review by ijuma — APPROVED** [review]
Thanks for the PR, just a nit.

File: build.gradle (hunk 2 above)
Comment (rajinisivaram) [reply build.gradle]: @ijuma Thanks for the review, updated.

**Review by ijuma — APPROVED** [review]
LGTM

File: build.gradle (hunk 1 above)
Comment (guozhangwang) [reply build.gradle]: I think we have a separate jar for each scala version as well for kafka-streams-scala?

File: release.py [hunk 3]
```diff
@@ -495,7 +495,6 @@ def select_gpg_key():
 if not user_ok("Going to build and upload mvn artifacts based on these settings:\n" + contents + '\nOK (y/n)?: '):
     fail("Retry again later")
 cmd("Building and uploading archives", "./gradlew uploadArchivesAll", cwd=kafka_dir, env=jdk8_env)
-cmd("Building and uploading archives", "./gradlew uploadCoreArchives_2_12 -PscalaVersion=2.12", cwd=kafka_dir, env=jdk8_env)
```
Comment (guozhangwang) [inline release.py:498]: Just curious: seems we do not need this step even before this PR. Is that right?

**Review by guozhangwang — COMMENTED** [review]
The patch LGTM.

File: release.py (hunk 3 above)
Comment (ijuma) [reply release.py:498]: Yes, that should have been removed in a previous PR.

File: build.gradle (hunk 1 above)
Comment (rajinisivaram) [reply build.gradle]: @ijuma @guozhangwang Sorry, this was an issue with the older version of gradle that I was testing with. Updated that and now it does generate separate source jars (I dont understand why). Updated `build.gradle`.

**Review by ijuma — APPROVED** [review]
LGTM

**Comment by rajinisivaram** [comment]
retest this please

**Comment by rajinisivaram** [comment]
Merging to trunk and 2.0.

**Pull request merged**